source/frontend/src/features/documents/list/DocumentActivityListContainer.tsx

```diff
@@ -0,0 +1,129 @@
+import { exists, getApiPropertyName, relationshipTypeToPathName } from '@/utils';
+
+import { DocumentRow } from '../ComposedDocument';
+import { useDocumentRelationshipProvider } from '../hooks/useDocumentRelationshipProvider';
+import { IDocumentListContainerProps } from './DocumentListContainer';
+import DocumentListView from './DocumentListView';
+import { DocRelation } from './models';
+
+const DocumentActivityListContainer: React.FunctionComponent<
+  IDocumentListContainerProps
+> = props => {
+  const pathGenerator = usePathGenerator();
+
+  const isMounted = useIsMounted();
+
+  const [documentResults, setDocumentResults] = useState<DocumentRow[]>([]);
+
+  const [activity, setActivity] = useState<ApiGen_Concepts_PropertyActivity>(null);
+
+  const {
+    getPropertyManagementActivity: { execute: getManagementActivity, loading: getActivityLoading },
+  } = useManagementActivityRepository();
+
+  const {
+    getAllPropertiesById: { execute: getProperties, loading: getPropertiesLoading },
+  } = usePimsPropertyRepository();
+
+  const { retrieveDocumentRelationship, retrieveDocumentRelationshipLoading } =
+    useDocumentRelationshipProvider();
+
+  const retrieveActivities = useCallback(async () => {
+    const result = await getManagementActivity(Number(props.parentId));
+    if (exists(result) && isMounted()) {
+      const propertyResult = await getProperties(
+        result.activityProperties.map(ap => ap.propertyId),
+      );
+      result.activityProperties.forEach(ap => {
+        const foundProperty = propertyResult.find(p => p.id === ap.propertyId);
+        ap.property = foundProperty;
+      });
+      setActivity(result);
+    }
+  }, [getManagementActivity, props.parentId, isMounted, getProperties]);
+
+  const retrieveActivitiesDocuments = useCallback(
+    async (activity: ApiGen_Concepts_PropertyActivity) => {
+      if (!exists(activity)) {
+        return;
+      }
+
+      const docRelations: DocRelation[] = activity?.activityProperties?.map(x => ({
+        id: x.propertyId,
+        relationType: ApiGen_CodeTypes_DocumentRelationType.Properties,
+        fileNumber: getApiPropertyName(x.property).value,
+      }));
+
+      let documentRows: DocumentRow[] = [];
+      for (const docRelation of docRelations.filter(exists)) {
+        const documents = await retrieveDocumentRelationship(
+          docRelation.relationType,
+          docRelation.id.toString(),
+        );
+        if (documents !== undefined && isMounted()) {
+          documentRows = documentRows.concat(
+            documents
+              .filter((x): x is ApiGen_Concepts_DocumentRelationship => !!x?.document)
+              .map(x => DocumentRow.fromApi(x, docRelation.fileNumber)),
+          );
+        }
+      }
+      setDocumentResults([...documentRows]);
+    },
+    [retrieveDocumentRelationship, isMounted],
+  );
+
+  useEffect(() => {
```

Comment:
Can you merge these useEffects? it seems like the intent is to never run effect 1 without effect 2.